### PR TITLE
Shuffle node embeddings according to node_id_mapping when saving node embeddings

### PR DESCRIPTION
**Pull request by classicsong**
*Issue #, if available:*
#49 

*Description of changes:*
If node id mapping exists in the partition files, shuffle node embeddings according to the node_id_mapping when save node embeddings.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/unit-tests/test_utils.py [hunk 1]
```diff
@@ -55,6 +66,134 @@ def helper_save_embedding(tmpdirname):
...
+# Will add test for nccl once we enable nccl
+@pytest.mark.parametrize("num_embs", [16, 17])
+@pytest.mark.parametrize("backend", ["gloo"])
+def test_save_embeddings_with_id_mapping(num_embs, backend):
+    import tempfile
+
+    # single embedding
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        emb, nid_mapping = gen_embedding_with_nid_mapping(num_embs)
+        print(nid_mapping)
```
Comment (zhjwy9343) [inline tests/unit-tests/test_utils.py:124]: Do we need to print the nid_mapping?

**Review by zhjwy9343 — APPROVED** [review]
LGTM, except for a few wording issues.

**Review by zhjwy9343 — APPROVED** [review]
LGTM, except for a few wording issues.

**Comment by zhjwy9343** [comment]
The new argument is not added to argument.py and cause the failure of end-to-end test.

File: tests/unit-tests/test_utils.py (hunk 1 above)
Comment (classicsong) [reply tests/unit-tests/test_utils.py:124]: NO.

File: python/graphstorm/model/utils.py [hunk 2]
```diff
@@ -250,33 +289,103 @@ def save_embeddings(model_path, embeddings, local_rank, world_size):
...
     assert local_rank < world_size
-    def get_data_range(num_embs):
-        # Get corresponding data range
-        start = local_rank * (num_embs // world_size)
-        end = (local_rank + 1) * (num_embs // world_size)
-        end = num_embs if local_rank + 1 == world_size else end
-        return start, end
+    backend = th.distributed.get_backend()
+    device = th.device('cpu') if backend == "gloo" else device
+    def exchange_node_id_mapping(node_id_mapping, num_embs):
```
Comment (zheng-da) [inline python/graphstorm/model/utils.py:299]: it's better to move the function out. it's not easy to read. it also makes the main function too long.

**Review by zheng-da — APPROVED** [review]

File: python/graphstorm/model/utils.py (hunk 2 above)
Comment (classicsong) [reply python/graphstorm/model/utils.py]: Updated

**Pull request merged**